lua/powerups/server/groundpound.moon

```diff
@@ -0,0 +1,355 @@
+                    \EmitSound "ambient/machines/thumper_dust.wav", 90, 62, 1, CHAN_STATIC 
+
+                if speedClamped > TERMINAL_VELOCITY * 0.35 -- medium speed hit
+                    hitPitch = 175 + -( speedZeroToOne * 125 )
+                    hitLvl = 90 + ( speedZeroToOne * 50 )
+                    \EmitSound "weapons/mortar/mortar_explode2.wav", hitLvl, hitPitch
+
+                if speedClamped > TERMINAL_VELOCITY * 0.65 -- high speed hit
+                    crashPitch = 110 + -( speedZeroToOne * 40 )
+                    crashLvl = 85 + ( speedZeroToOne * 60 )
+                    \EmitSound "ambient/machines/wall_crash1.wav", crashLvl, crashPitch, 0.5
+
+                if speedClamped > TERMINAL_VELOCITY * 0.9 -- all powerful hit!
+                    \EmitSound "ambient/explosions/exp2.wav", 110, 30, 0.5 -- echo
+                    util.BlastDamage @damageInflictor, @owner, ownersPos, radius * TERMINAL_EXTRABLAST_RADIUS_MUL, damage * TERMINAL_EXTRABLAST_DMGMUL -- OP blastdamage in a tiny, tiny radius
+
+            timer.Simple 0, -> -- prediction...
+                return unless IsValid @owner
+
+                if speedClamped > TERMINAL_VELOCITY * 0.9
+                    effDat = EffectData!
+                    with effDat
+                        \SetOrigin ownersPos
+                        \SetEntity @owner
+
+                        \SetScale 1
+                        util.Effect "powerups_groundpound_shockwave", effDat
+
+                        \SetScale 4
+                        \SetNormal UP_VECTOR
+                        util.Effect "powerups_groundpound_shockwave_huge", effDat
+
+                else
+                    effDat = EffectData!
+                    effScale = speedZeroToOne * 0.5
+                    effScale = effScale + 0.2
+                    with effDat
+                        \SetOrigin ownersPos
+                        \SetNormal UP_VECTOR
+                        \SetScale effScale
+                        util.Effect "powerups_groundpound_shockwave", effDat
+
+            if @UsesRemaining < 1
+                @Remove!
+
+            return 0
+
+    CreateThinkWatcher: =>
+        () ->
+            owner = @owner
+
+            checkCantFastFall = () ->
+                return true unless owner\Alive!
+                return true if owner\InVehicle!
+                return true if owner\IsOnGround!
+                return true if BAD_MOVETYPES[owner\GetMoveType!]
+                return true unless owner\KeyDown IN_DUCK
+
+                return false
+            
+            cantFastFall = checkCantFastFall!
+
+            if cantFastFall
+                if @fastFalling
+                    @fastFalling = false
+                    @upToSpeedSound = false
+                    @upToTerminalSpeedSound = false
+                    @fallSound\ChangeVolume 0, FALL_SOUND_FADE_OUT
+
+                return
```

Comment:
```suggestion
            if cantFastFall
                return unless @fastFalling

                @fastFalling = false
                @upToSpeedSound = false
                @upToTerminalSpeedSound = false
                @fallSound\ChangeVolume 0, FALL_SOUND_FADE_OUT
```

Reply:
That would change the logic. I want it to always return if `cantFastFall` is true, but *also* change those vars if it was previously fast falling.